crt_portal/cts_forms/page_through.py

```diff
@@ -0,0 +1,53 @@
+from django.core.paginator import EmptyPage, PageNotAnInteger
+
+
+def pagination(paginator, page, per_page):
+    page = int(page)
+    try:
+        records = paginator.page(page)
+    except PageNotAnInteger:
+        records = paginator.page(1)
+    except EmptyPage:
+        records = paginator.page(paginator.num_pages)
+
+    adjacent_pages = 1
+
+    show_first = page > adjacent_pages + 1
+    first_ellipsis = show_first and not (page <= adjacent_pages + 2)
+
+    show_last = page < paginator.num_pages - adjacent_pages
+    last_ellipsis = show_last and not (paginator.num_pages - adjacent_pages - 1)
```

Comment:
@LindsayYoung Should this line be as follows?:

```
last_ellipsis = show_last and not (page == paginator.num_pages - adjacent_pages - 1)
```

Reply:
Yes, thank you!! I can update in the next 20 minutes or so